### PR TITLE
fix(focus-trap): update focus trap attrs to camel case #6799

### PR DESCRIPTION
**Pull request by amcdnl**
Addresses #6799

### CODE REVIEW

File: src/cdk/a11y/focus-trap.ts [hunk 1]
```diff
@@ -144,19 +144,24 @@ export class FocusTrap {
    * @param bound The boundary to get (start or end of trapped region).
    * @returns The boundary element.
    */
-  private _getRegionBoundary(bound: 'start' | 'end'): HTMLElement | null {
+  private _getRegionBoundary(bound: 'Start' | 'End'): HTMLElement | null {
```
Comment (mmalerba) [inline src/cdk/a11y/focus-trap.ts]: Can we change this back to `'start'` and `'end'`? I'm pretty sure everywhere else we use string literal types we use all lowercase, we can just convert to upper case in the method

Comment (amcdnl) [reply src/cdk/a11y/focus-trap.ts]: done :)

File: src/demo-app/sidenav/sidenav-demo.html [hunk 2]
```diff
@@ -68,10 +68,10 @@
   <md-sidenav #focusSidenav>
     <md-nav-list>
       <a md-list-item routerLink>Link</a>
-      <a md-list-item routerLink cdk-focus-region-start>Focus region start</a>
+      <a md-list-item routerLink cdkFocusStart>Focus region start</a>
```
Comment (mmalerba) [inline src/demo-app/sidenav/sidenav-demo.html]: `cdkFocusRegionStart`

File: src/demo-app/sidenav/sidenav-demo.html [hunk 3]
```diff
@@ -68,10 +68,10 @@
...
       <a md-list-item routerLink>Link</a>
-      <a md-list-item routerLink cdk-focus-initial>Initially focused</a>
-      <a md-list-item routerLink cdk-focus-region-end>Focus region end</a>
+      <a md-list-item routerLink cdkFocusInitial>Initially focused</a>
+      <a md-list-item routerLink cdkFocusEnd>Focus region end</a>
```
Comment (mmalerba) [inline src/demo-app/sidenav/sidenav-demo.html]: `cdkFocusRegionEnd`

Comment (amcdnl) [reply src/demo-app/sidenav/sidenav-demo.html]: * face palm * ---> fixed.

**Comment by mmalerba** [comment]
please rebase

**Comment by amcdnl** [comment]
@mmalerba - done. Noticed that the focus trap is removed from the demo now, is that by design?

**Comment by mmalerba** [comment]
@amcdnl it just moved to the drawer demo: https://github.com/angular/material2/blob/master/src/demo-app/drawer/drawer-demo.html

**Comment by mmalerba** [comment]
@amcdnl rebase?

**Comment by amcdnl** [comment]
@mmalerba - done :)

**Comment by mmalerba** [comment]
And the sidenav demo got moved to the drawer demo, so that needs update as well

**Comment by amcdnl** [comment]
@mmalerba - done

File: src/cdk/a11y/focus-trap.ts [hunk 4]
```diff
@@ -148,15 +148,19 @@ export class FocusTrap {
     if (!this._platform.isBrowser) {
       return null;
     }
-
+    
     // Contains the deprecated version of selector, for temporary backwards comparability.
     let markers = this._element.querySelectorAll(`[cdk-focus-region-${bound}], ` +
-                                                 `[cdk-focus-${bound}]`) as NodeListOf<HTMLElement>;
+                                                 `[cdkFocusRegion${bound}], ` +
+                                                 `[cdk-focus-${name}]`) as NodeListOf<HTMLElement>;
```
Comment (mmalerba) [inline src/cdk/a11y/focus-trap.ts]: `name` should be `bound`

File: src/cdk/a11y/focus-trap.ts [hunk 5]
```diff
@@ -148,15 +148,19 @@ export class FocusTrap {
...
 
     for (let i = 0; i < markers.length; i++) {
       if (markers[i].hasAttribute(`cdk-focus-${bound}`)) {
         console.warn(`Found use of deprecated attribute 'cdk-focus-${bound}',` +
-                     ` use 'cdk-focus-region-${bound}' instead.`, markers[i]);
+                     ` use 'cdkFocusRegion${name}' instead.`, markers[i]);
```
Comment (mmalerba) [inline src/cdk/a11y/focus-trap.ts]: `name` should be `bound`

File: src/cdk/a11y/focus-trap.ts [hunk 6]
```diff
@@ -148,15 +148,19 @@ export class FocusTrap {
...
+      } else if (markers[i].hasAttribute(`cdk-focus-region-${bound}`)) {
+        console.warn(`Found use of deprecated attribute 'cdk-focus-region-${bound}',` +
+                     ` use 'cdkFocusRegion${name}' instead.`, markers[i]);
```
Comment (mmalerba) [inline src/cdk/a11y/focus-trap.ts]: `name` should be `bound`

**Comment by amcdnl** [comment]
Done :)

**Pull request merged**